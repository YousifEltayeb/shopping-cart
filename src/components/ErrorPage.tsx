import { Link } from "react-router";
import styled from "styled-components";

const Main = styled.main``;
const ErrorPage = () => {
  return (
    <Main aria-label="ErrorPage">
      <h1>Oh no, this route does not exist!</h1>
      <Link to="/">
        You can go back to the home page by clicking here, though!
      </Link>
    </Main>
  );
};

export default ErrorPage;
