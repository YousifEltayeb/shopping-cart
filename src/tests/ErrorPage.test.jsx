import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import ErrorPage from "../components/ErrorPage";
import { MemoryRouter } from "react-router-dom";
describe("ErrorPage", () => {
    it("Error message is rendered", () => {
        const { container } = render(
            <MemoryRouter>
                <ErrorPage />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
