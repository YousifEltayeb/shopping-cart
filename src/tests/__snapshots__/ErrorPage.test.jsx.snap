// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > Error message is rendered 1`] = `
<div>
  <div>
    <h1>
      Oh no, this route does not exist!
    </h1>
    <a
      data-discover="true"
      href="/"
    >
      You can go back to the home page by clicking here, though!
    </a>
  </div>
</div>
`;
